### PR TITLE
feat: update operator to update a table

### PR DESCRIPTION
**Pull request by scgkiran**

### CODE REVIEW

File: proto/substrait/algebra.proto [hunk 1]
```diff
@@ -619,7 +620,7 @@ message WriteRel {
   // The type of operation to perform
   WriteOp op = 4;
 
-  // The relation that determines the records to add/remove/modify
```
Comment (EpsilonPrime) [inline proto/substrait/algebra.proto:622]: Modification using this particular method is still reasonable -- one would use this if you had ids to merge against.

File: site/docs/relations/logical_relations.md [hunk 2]
```diff
@@ -474,6 +474,32 @@ Write definition types are built by the community and added to the specification
 | Format   | Enumeration of available formats. Only current option is PARQUET. | Required |
 
 
+## Update Operator
+
+The update operator is an operator that consumes one input and writes the column updates to a storage. 
+
+| Signature            | Value                                 |
+| -------------------- |---------------------------------------|
+| Inputs               | 1                                     |
```
Comment (EpsilonPrime) [inline site/docs/relations/logical_relations.md:483]: Technically this new operator doesn't have an input (besides the table name).

File: proto/substrait/algebra.proto (hunk 1 above)
Comment (scgkiran) [reply proto/substrait/algebra.proto:622]: In current state it is not possible. Right?
We can modify the comment if we modify this proto.

File: site/docs/relations/logical_relations.md (hunk 2 above)
Comment (scgkiran) [reply site/docs/relations/logical_relations.md:483]: We can consider the table name on which the operator applied as input?
0 inputs for a update operator sounds strange.

File: proto/substrait/algebra.proto (hunk 1 above)
Comment (scgkiran) [reply proto/substrait/algebra.proto:622]: ok got it, will revert the changes related to WriteRel.

File: site/docs/relations/logical_relations.md (hunk 2 above)
Comment (EpsilonPrime) [reply site/docs/relations/logical_relations.md:483]: Inputs are traditionally other relations.  You could say zero and add a comment explaining that it does get data from somewhere.  The closest analog we have for existing relations is ReadRel.

Comment (scgkiran) [reply site/docs/relations/logical_relations.md]: ok fixed it. Please take a look.

**Review by EpsilonPrime — APPROVED** [review]

**Review by jacques-n — APPROVED** [review]

**Pull request merged**